lib/fastifyCsrf.d.ts

```diff
@@ -1,37 +1,32 @@
 /// <reference types="node" />
-import * as fastify from 'fastify';
-import { Options as TokenOptions } from 'csrf';
-import 'fastify-cookie';
 
-declare module 'fastify' {
-    interface FastifyRequest<HttpRequest,
-        Query = fastify.DefaultQuery,
-        Params = fastify.DefaultParams,
-        Headers = fastify.DefaultHeaders,
-        Body = any> {
-        csrfToken(): string;
-    }
-}
+import * as http from "http";
+import * as fastify from "fastify";
+import { Options as TokenOptions } from "csrf";
+import "fastify-cookie";
 
-declare function FastifyCsrfPlugin<
-    HttpServer = any,
-    HttpRequest = any,
-    HttpResponse = any>(opts?: FastifyCsrfPlugin.Options<HttpRequest>): FastifyCsrfPlugin.Plugin<
-    HttpServer,
-    HttpRequest,
-    HttpResponse>;
+type HttpServer = http.Server;
+type HttpRequest = http.IncomingMessage;
+type HttpResponse = http.ServerResponse;
 
-declare namespace FastifyCsrfPlugin {
-    interface Options<HttpRequest> extends TokenOptions {
-        value?: (req: HttpRequest) => string;
-        cookie?: fastify.CookieSerializeOptions | boolean;
-        ignoreMethods?: string[];
-        sessionKey?: string;
-    }
+declare module "fastify" {
+  interface FastifyRequest {
+    csrfToken(): string;
+  }
+}
 
-    interface Plugin<HttpServer, HttpRequest, HttpResponse>
-        extends fastify.Plugin<HttpServer, HttpRequest, HttpResponse, FastifyCsrfPlugin.Options<HttpRequest>> {
-    }
+interface Options extends TokenOptions {
+  key?: string;
+  value?: (req: fastify.FastifyRequest) => string;
+  cookie?: fastify.CookieSerializeOptions | boolean;
+  ignoreMethods?: Array<string>;
 }
 
-export = FastifyCsrfPlugin;
+declare const fastifyCsrf: fastify.Plugin<
```

Comment:
The `fastify.Plugin` type accepts only the `Options` type parameter, see: https://github.com/fastify/fastify/blob/master/types/plugin.d.ts#L10

This also makes `HttpServer`, `HttpRequest` and `HttpResponse` obsolete, so they can be removed from this file completely.

Reply:
Yeah, I thought so too at first, but since we currently depend on `fastify@2.x` we need to look at https://github.com/fastify/fastify/blob/2.x/fastify.d.ts#L26 and there we the typings are less sophisticated unfortunately 😐

So should we keep it the way it is for now?

Reply:
Oh, my bad! Thanks for pointing this out :+1: .

Please keep it the way it is.